### PR TITLE
chore: Remove more references to timberio

### PR DESCRIPTION
**Pull request by jszwedko**
I think the only references left now are for docker images which require
us to switch over our Dockerhub org before we can update.

Closes: #11103

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: .github/semantic.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # WARNING! THIS FILE IS AUTOGENERATED! DO NOT EDIT!
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply .github/semantic.yml:1]: Doh, I didn't even open it 😅 This one was `sed`'d.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ further defined and clarified by project maintainers.
 ## Enforcement
 
 Instances of abusive, harassing, or otherwise unacceptable behavior may be
-reported by contacting the project team at [vector@timber.io](mailto:vector@timber.io).
+reported by contacting the project team at [vector@datadoghq.com](mailto:vector@datadoghq.com).
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply CODE_OF_CONDUCT.md:70]: I'm actually trying to see if I can make this `team@vector.dev`, now, instead. I'll move this back to draft until that's decided.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply CODE_OF_CONDUCT.md:70]: FWIW I have the helm charts with `team-vector@datadoghq.com` so we could update there if we got another address

[the remark this answers is not in the document]
Comment (jszwedko) [reply CODE_OF_CONDUCT.md:70]: `team@vector.dev` is still TBD, but I was able to get `vector@datadoghq.com` setup so we can use that one for now. That one allows external email. `team-vector@datadoghq.com` does not.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply CODE_OF_CONDUCT.md:70]: OK, didn't know that... so I'll open a PR on the charts

**Pull request merged**